### ISSUE TITLE
⁉️ Editing of Server-Side Diagrams Possible

### ISSUE THREAD

**Issue by nox-do** [process-engine/bpmn-studio#1038]
#### Describe your issue
It seems to subject to irritation and problematic if a user can edit a remote (uploaded) diagram. Personally I think:

a) If I do not have the diagram locally why should I change it remotely? Potentially not "mine" in case of collaboration.
b) IF the server side differs, how can I sync (or download) to local?

c) A little out of scope - How do I delete a remote diagram? 

#### How can others reproduce the issue?
- Draw a diagram and upload it
- Modify it

#### Possible solution
It's general question of how it's supposed to be.  (Allow download, deny editing on remote) or allow with warning.

But beside this at least I'd like to have the option to delete / download  a remote process. 

#### What's your setup
- BPMN-Studio Version (`4.6.0.x`):
- OS (`Windows`):

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

**Comment by cmg-dev** [comment]
I just have time for a short answer, a long one will follow :)

> c) A little out of scope - How do I delete a remote diagram?

currently this is not possible. Due to the "old" way of directly accessing the data layer, we did not like this.

Long Story short: After splitting Persistence and Storage APIs, this will be possible again.
Please follow this issue: https://github.com/process-engine/process_engine_runtime/issues/84

This is the *start* of this story.

**Comment by cmg-dev** [comment]
> b) IF the server side differs, how can I sync (or download) to local?

There will be a section in the Documentation Mk IV to explain this.
I have to admit, it is quite cumbersome.

But you can:

1. Open the diagram from the ProcessEngine.
1. Click Export as BPMN.
1. Save it over the local diagram.
1. The Solution Explorer will reload the local diagram.

This is something I will definitely explain in the Documentation.

**Comment by nox-do** [comment]
+1

**Comment by cmg-dev** [comment]
> a) If I do not have the diagram locally why should I change it remotely? Potentially not "mine" in case of collaboration.

**TL;DR:**

You are right; this happens, because we have super user privileges in the BPMN-Studio and it's defaulting ProcessEngine.

**Long Version:**

This is something, were the beauty of the IAM-- with OAuth/OpenID-- comes into play. It boils down to this call in the ProcessEngine:

https://github.com/process-engine/process_engine_core/blob/develop/src/runtime/persistence/process_model_service.ts#L24

If: 

1. a given Identity has this claim, it is able to update-- AKA Deploy-- the diagram.
1. it would not have this claim, it will be denied the writing.

We:

1. use the `dummy_token`-Mechanism, that will grant us a 'GOD_MODE' privileges in the BPMN-Studio.
1. use this, because we want an easy access for the first time BPMN-Studio users.
1. do not recommend this in a production setup.
1. can restrain the access to deployed diagrams, by using a configured Authority. 

So,:

1. DO NOT USE SUPER USER PRIVILEGES IN PRODUCTION ;)
1. a configured Identity-Provider (like: https://github.com/process-engine/identity_server) will allow to specify, who can deploy/write to diagram.
1. as writing is protected by: https://github.com/process-engine/process_engine_core/blob/develop/src/runtime/persistence/process_model_service.ts#L24 it is easy to configure the claim.

Hope this will help you.

\cc @Vyperus

**Comment by rrrene** [comment]
Perspektivisch: 

* lokale ProcessEngine lässt Editieren zu
* auf Remote-ProcessEngines erlaubt BPMN Studio dies zukünftig per default nicht

**Comment by alexanderkasten** [comment]
> Perspektivisch:
> 
> * lokale ProcessEngine lässt Editieren zu
> * auf Remote-ProcessEngines erlaubt BPMN Studio dies zukünftig per default nicht

Seit dem Release v5.3.0 erledigt.
